viser/_message_api.py

```diff
@@ -341,6 +341,42 @@ def add_mesh_trimesh(
 
         return MeshHandle._make(self, name, wxyz, position, visible)
 
+    def set_popup_image(
+        self,
+        image: onp.ndarray,
+        depth_img: onp.ndarray[float] ,
+        format: Literal["png", "jpeg"] = "jpeg",
+        jpeg_quality: Optional[int] = None,
+    ) -> None:
+        """
+        Set a popup image for the scene. Useful for NeRF visualization.
+        
+        Expects depth_img to be a HxWx1 array of depth values in **floating point**
+        """
+        assert depth_img.dtype == onp.float32 or depth_img.dtype == onp.float16 or depth_img.dtype == onp.float64
+        assert len(depth_img.shape) == 3 
+        assert depth_img.shape[2] == 1
+        media_type, base64_rgb = _encode_image_base64(
+            image, format, jpeg_quality=jpeg_quality
+        )
+        base = 10.0
+        # convert to fixed point float, with precision of .01 ranging from .01 to 99.99
```

Comment:
is this throwing away a lot of bits? like each byte can store `[0, 255]` but we only use `[0, 9]`?

Reply:
It's storing the range [0,9] as 255 discrete steps I think, since (x/10.0)*255 where x \in (0,9) maps to [0,255]. The sampler on the other side samples a continuous value which gets unpacked by scaling by 10.